frontend/ci-plugins-generator.js

```diff
@@ -0,0 +1,71 @@
+const path = require('node:path');
+const fs = require('node:fs');
+const _ = require("lodash");
+
+const LINKED_PLUGINS_MODULE_TEMPLATE = (plugins) => {
+  const importableName = (name) => _.upperFirst(_.camelCase(name));
+  const frontendPlugins = plugins.map(([name, _]) => [name, importableName(name)]);
+
+  return `
+import {NgModule} from "@angular/core";
+${
+  frontendPlugins
+    .map(([actualName, moduleName]) => 
+      `import {PluginModule as ${moduleName}} from './linked/${actualName}/main';`
+    )
+    .join("\n")
+}
+
+@NgModule({
+    imports: [
+        ${
+          frontendPlugins
+            .map(([_, moduleName]) => moduleName)
+            .join(`,\n${" ".repeat(8)}`)
+        }
+    ],
+})
+export class LinkedPluginsModule { }
+  `;
+};
+
+const railsRout = path.join(__dirname, "..");
+const pluginDir = path.join(railsRout, 'modules');
+const targetDir = path.join(railsRout, 'frontend/src/app/features/plugins/linked');
+
+const plugins = new Map([
+  ["budgets", path.join(pluginDir, "budgets")],
+  ["costs", path.join(pluginDir, "costs")],
+  ["openproject-avatars", path.join(pluginDir, "avatars")],
+  ["openproject-documents", path.join(pluginDir, "documents")],
+  ["openproject-github_integration", path.join(pluginDir, "github_integration")],
+  ["openproject-gitlab_integration", path.join(pluginDir, "gitlab_integration")],
+  ["openproject-meeting", path.join(pluginDir, "meeting")]
+]);
+
+console.log(`Cleaning linked target directory ${targetDir}`);
+fs.rmSync(targetDir, { recursive: true, force: true });
+fs.mkdirSync(targetDir);
+
+plugins.forEach((pluginPath, name) => {
+  const linkTarget = path.join(pluginPath, "frontend", "module");
+  const linkPath = path.join(targetDir, name);
+
+  console.log(`Linking frontend of OpenProject plugin ${name} to ${linkPath}.`)
+  fs.symlinkSync(linkTarget, linkPath);
+});
+
+const allFrontendPlugins = Array.from(plugins).filter(([_, pluginPath]) => {
+  const frontendEntry = path.join(pluginPath, "frontend", "module", "main.ts");
+  return fs.existsSync(frontendEntry);
+});
+
+function generatePluginModule(plugins) {
+  const fileRegister = path.join(railsRout, "frontend/src/app/features/plugins/linked-plugins.module.ts");
```

Comment:
This code duplicates the plugin linking of https://github.com/opf/openproject/blob/4911b8a14921efe0c552ab2029389ac12bcaefcb/lib/open_project/plugins/frontend_linking.rb#L35, so it will not test the reality of the build.

We can't get rid of the ruby-based building, as plugins might be added by the Gemfile. I'd vote that we either

- use setup-ruby, bundle with cache und use the actual plugin linking code
- accept that plugins are not properly tested in the frontend code